### DISCUSSION TITLE
Add Headers to Requests  executed by  WebHooks

### DISCUSSION THREAD

**Discussion by LuchitoPTY** [directus/directus#3649] [Feature Requests]
Hello Directus Team!

It is necessary in some cases add a security header in order to call a Rest api endpoint.
Example: Invoke a build on AzureDevops using a WebHook as build trigger.

App UI should has a field to add headers to the request URL.

**Comment by emregency** [comment]
Hi @LuchitoPTY,

Same concern here #3000.

I am not using the Azure DevOps flow but I use Azure Functions. I had a similar challenge, and functions allow to pass it as query params `?code=`. Maybe you want to check if you have the option to use a similar method.

**Comment by ghost** [comment]
I have the same issue with Directus triggering builds on DroneCI. I made this little guy to get around it: https://git.atrilahiji.dev/atrilahiji/MiddleWhere

**Reply by iba-1** [reply]
Same thing applies to Github Action: t[o trigger a workflow using repository dispatch ](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#repository_dispatch) you need to send Authorization and Accept headers, plus a body if you want to trigger workflow based on some event_type.
We are actually using NGINX to rewrite the request and send it to Github, but we would love to have this feature in Directus!

**Reply by cupcakearmy** [reply]
@rijkvanzanten 

Is there a ticket for that? I'm having the same issues. Want to trigger github actions but without headers that's kind of impossible

**Reply by rijkvanzanten** [reply]
@cupcakearmy I believe there's an open PR that just came in actually! https://github.com/directus/directus/pull/8855

**Comment by superhp** [comment]
To have fields for authentication and message body are absolutely needed features. I see this has been requested for quite some time. Are there any plans to fix it? Or maybe someone has implemented an extension hook for this and could share it?

**Reply by rijkvanzanten** [reply]
The webhook already sends a body with the webhook payload. For any more specific usage or custom headers or anything else, use a custom hook https://docs.directus.io/guides/api-hooks/ to send your own http request 👍🏻

**Reply by superhp** [reply]
It can send the body, but not the custom body which could be entered manually. I'm planning to implement the api hook for that as you suggested. But do you have any plans to make it more flexible from UI side as well?

**Reply by rijkvanzanten** [reply]
If there's enough community demand for it, sure! By supporting every custom bit you can add to a request when you do it manually in code in the UI as well, we make the UI so complicated that it might become easier to do it in code anyways, so we'll have to find the balance in "just works" ease of use vs power-user configuration flexibility

**Reply by ghost** [reply]
Thank you for this information @rijkvanzanten I am going to have to look at custom hooks as I was also looking to do this through UI. Interesting points on the design philosophy in last reply

**Comment by rijkvanzanten** [comment]
This has since been added through Flows!
